### PR TITLE
JobSet default managed by

### PR DESCRIPTION
**Pull request by vladikkuzn**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Don't reject JobSet if cluster queue doesn't exist, make it suspended
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/kueue/issues/2416

#### Does this PR introduce a user-facing change?

```release-note
JobSet is not rejected if cluster queue doesn't exist, now it's suspended
```

### CODE REVIEW

**Comment by trasc** [comment]
/hold

for #2401

**Comment by mimowo** [comment]
> /hold
> 
> for #2401

Actually, the idea is to merge the fix first, then follow up with Job: https://github.com/kubernetes-sigs/kueue/pull/2401#discussion_r1642835775

File: pkg/controller/jobs/jobset/jobset_webhook.go [hunk 1]
```diff
@@ -73,14 +73,15 @@ func (w *JobSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 		if !found {
 			return nil
 		}
-		clusterQueueName, err := w.queues.ClusterQueueFromLocalQueue(queue.QueueKey(jobSet.ObjectMeta.Namespace, localQueueName))
-		if err != nil {
-			return err
+		clusterQueueName, ok := w.queues.ClusterQueueFromLocalQueue(queue.QueueKey(jobSet.ObjectMeta.Namespace, localQueueName))
+		if !ok {
+			log.V(5).Info("Cluster queue for local queue not found", "jobset", klog.KObj(jobSet), "localQueue", localQueueName)
```
Comment (mimowo) [inline pkg/controller/jobs/jobset/jobset_webhook.go:78]:
```suggestion
			log.V(5).Info("Cluster queue for local queue not found", "jobset", klog.KObj(jobSet), "localQueue", localQueueName, "clusterQueue", clusterQueueName)
```

**Review by mimowo — COMMENTED** [review]
/approve
LGTM, just one small nit

File: pkg/controller/jobs/jobset/jobset_webhook.go (hunk 1 above)
Comment (mimowo) [reply pkg/controller/jobs/jobset/jobset_webhook.go:78]: ah, sure, thanks

**Comment by mimowo** [comment]
/release-note-edit
```release-note
Do not reject a JobSet if the corresponding cluster queue doesn't exist (when MultiKueue is enabled).
```
/remove-kind feature
/kind bug

**Comment by mimowo** [comment]
/retitle "[MultiKueue] Do not reject JobSet when ClusterQueue does not exist"

**Comment by trasc** [comment]
> > /hold
> > for #2401
> 
> Actually, the idea is to merge the fix first, then follow up with Job: [#2401 (comment)](https://github.com/kubernetes-sigs/kueue/pull/2401#discussion_r1642835775)

Especially if cherry-picked, the fix should be minimal, there are parts of this pr that are not needed to not reject the JobSet if the queue is missing.

**Comment by mimowo** [comment]
> Especially if cherry-picked, the fix should be minimal, there are parts of this pr that are not needed to not reject the JobSet if the queue is missing.

Are you talking about the change of the `ClusterQueueFromLocalQueue` returned type (bool vs error)? I think this is acceptable.

**Comment by trasc** [comment]
> > Especially if cherry-picked, the fix should be minimal, there are parts of this pr that are not needed to not reject the JobSet if the queue is missing.
> 
> Are you talking about the change of the `ClusterQueueFromLocalQueue` returned type (bool vs error)? I think this is acceptable.

That and the move of the controller name to API.

Besides pkg/controller/jobs/jobset/jobset_webhook.go:L78 and it's eventual test fix-ups nothing should have been done in this PR.

**Comment by mimowo** [comment]
I think the change of the function signature is ok. Otherwise we would need to handle the error with Info logging, which is not ideal either. The move of the const to the API package is not necessary indeed. 

I'm ok to adjust the cherry-pick manually without the move.
@cc @alculquicondor any concerns to cherry-pick this PR?

**Comment by trasc** [comment]
> I think the change of the function signature is ok. Otherwise we would need to handle the error with Info logging, which is not ideal either. The move of the const to the API package is not necessary indeed.

But it has very little to do with the fix.

**Comment by mimowo** [comment]
/retitle [MultiKueue] Do not reject JobSet when ClusterQueue does not exist

**Comment by mimowo** [comment]
> But it has very little to do with the fix.

It is part of the PR in the main branch, so I would prefer not to do alternative implementation just for release-0.7.

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
MultiKueue: Do not reject a JobSet if the corresponding cluster queue doesn't exist (when MultiKueue is enabled).
```

**Comment by trasc** [comment]
> It is part of the PR in the main branch, so I would prefer not to do alternative implementation just for release-0.7.

Holding this PR until 2401 was merged wold have made this PR only about not rejecting the jobSet when the cq is missing,  what is in fact needed to fix the issue.

**Comment by alculquicondor** [comment]
Hence why it's important to do preparatory PRs separately.

Let's split out the API change from the cherry-pick so that we are consistent with our general practices.

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
MultiKueue: Do not reject a JobSet if the corresponding cluster queue doesn't exist
```

**Pull request merged**